CMakeLists.txt

```diff
@@ -43,18 +43,23 @@ if(WIN32)
   ament_environment_hooks("${BINARY_PATH_HOOK}" "${PYTHONPATH_HOOK}")
 else()
   set(LIBRARY_PATH_HOOK "${AMENT_PACKAGE_TEMPLATE_DIR}/environment_hook/library_path.${SHELL_EXT}")
-  # Collect machine architecture triplet for libraries using GNU install dirs
-  execute_process(COMMAND gcc -dumpmachine OUTPUT_VARIABLE MULTIARCH_TRIPLET OUTPUT_STRIP_TRAILING_WHITESPACE)
-  # register multiarch information for .dsv generation
-  if(APPLE)
-    set(LIBRARY_PATH_ENV_VAR "DYLD_LIBRARY_PATH")
-  else()
-    set(LIBRARY_PATH_ENV_VAR "LD_LIBRARY_PATH")
+  # enable C language so that a trycompile can determine what the
+  # anticipated libdir will be.
+  enable_language(C)
+  include(GNUInstallDirs)
+  if(NOT ${CMAKE_INSTALL_LIBDIR} STREQUAL "lib")
```

Comment:
Is there a negative consequence to removing this if-condition?

Reply:
The hooks should be tolerant to duplicates, so if we were to allow the hook to add "lib" a second time, it would probably be OK. This is just an optimization.

Reply:
thanks